sympy/ntheory/elliptic_curve.py

```diff
@@ -122,8 +124,8 @@ def points(self):
         all_pt = set()
         if char >= 1:
             for i in range(char):
-                congruence_eq = ((self._eq.lhs - self._eq.rhs).subs({self.x: i, self.z: 1}))
-                sol = polynomial_congruence(congruence_eq, char)
+                congruence_eq = self._eq.subs({self.x: i, self.z: 1})
```

Comment:
Could the rhs be something like `self._eq(i, self.y, 1)`?

Reply:
This is raising an error. For ex: `A = Poly(x**2*y + x**2 + x*y + x*z, x, y, z)` then ` A(1,y,1)` gave `UnificationFailed: can't unify ZZ with ZZ[y], given (y, z) generators`